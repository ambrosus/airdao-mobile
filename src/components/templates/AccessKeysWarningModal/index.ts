import { AccessKeysWarningModal } from './AccessKeysWarningModal';

export { AccessKeysWarningModal };
