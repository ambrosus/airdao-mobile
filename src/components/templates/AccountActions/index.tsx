import React from 'react';
import { Row } from '@components/base';
import { scale } from '@utils/scaling';
import { Swap } from './Swap';
import { Send } from './Send';
import { Receive } from './Receive';
import { Staking } from './Staking';
import { Token } from '@models';
import { AccountDBModel } from '@database';

interface AccountActionsProps {
  address: string;
  token?: Token;
  selectedAccount: AccountDBModel | null;
}
export const AccountActions = (props: AccountActionsProps) => {
  const { address, token } = props;
  return (
    <Row
      alignItems="center"
      justifyContent="center"
      style={{ columnGap: scale(40) }}
    >
      <Swap />
      <Send address={address} token={token} />
      <Receive address={address} />
      <Staking />
    </Row>
  );
};
