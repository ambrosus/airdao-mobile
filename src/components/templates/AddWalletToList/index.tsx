import React, { useMemo, useState } from 'react';
import { FlatList, ListRenderItemInfo, View } from 'react-native';
import { Button, Row, Spacer, Text } from '@components/base';
import { InputWithIcon } from '@components/composite';
import { verticalScale } from '@utils/scaling';
import { styles } from './styles';
import { ExplorerAccount } from '@models/Explorer';
import { AccountList } from '@models/AccountList';
import { COLORS } from '@constants/colors';
import { SearchIcon } from '@components/svg/icons';
import { NumberUtils } from '@utils/number';
import { useLists } from '@contexts/ListsContext';

interface AddWalletToListProps {
  wallet: ExplorerAccount;
  lists: AccountList[];
  onWalletToggled?: () => unknown;
}

export const AddWalletToList = (props: AddWalletToListProps): JSX.Element => {
  const { wallet, lists, onWalletToggled } = props;
  const { toggleAddressInList } = useLists((v) => v);
  const [searchText, setSearchText] = useState('');
  const filteredLists = useMemo(
    () =>
      searchText ? lists.filter((l) => l.name.includes(searchText)) : lists,
    [lists, searchText]
  );

  const toggleWalletInList = (list: AccountList) => {
    toggleAddressInList(wallet, list);
    if (typeof onWalletToggled === 'function') onWalletToggled();
  };

  const renderList = (args: ListRenderItemInfo<AccountList>) => {
    const { item: list } = args;
    const onPress = () => {
      toggleWalletInList(list);
    };

    return (
      <Button
        onPress={onPress}
        testID="AddWalletToList_Container"
        style={styles.item}
      >
        <Row alignItems="center" justifyContent="space-between">
          <View style={{ flex: 1 }}>
            <Row alignItems="center" justifyContent="space-between">
              <Text
                fontSize={14}
                fontFamily="Inter_500Medium"
                fontWeight="500"
                color={COLORS.smokyBlack}
              >
                {list.name}
              </Text>
              <Text
                fontSize={13}
                fontWeight="600"
                fontFamily="Mersad_600SemiBold"
                color={COLORS.smokyBlack}
              >
                ${NumberUtils.formatNumber(list.totalBalance, 2)}
              </Text>
            </Row>
            <Spacer value={verticalScale(4)} />
            <Text
              fontSize={12}
              fontWeight="500"
              fontFamily="Inter_500Medium"
              color={COLORS.smokyBlack50}
            >
              {list.accountCount} Addresses
            </Text>
          </View>
        </Row>
      </Button>
    );
  };

  return (
    <FlatList
      data={filteredLists}
      renderItem={renderList}
      keyExtractor={(l) => l.id}
      contentContainerStyle={styles.list}
      ListHeaderComponent={
        <View style={styles.searchBar}>
          <InputWithIcon
            value={searchText}
            onChangeValue={setSearchText}
            placeholder="Search collections"
            iconLeft={<SearchIcon color={COLORS.midnight} />}
          />
        </View>
      }
      stickyHeaderIndices={[0]}
    />
  );
};
