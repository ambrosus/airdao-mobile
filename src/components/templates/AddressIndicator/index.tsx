import React, { useMemo } from 'react';
import { View } from 'react-native';
import { styles } from './styles';
import { Text } from '@components/base';
import { COLORS } from '@constants/colors';
import { useWatchlist } from '@hooks';
import { WatchlistIcon } from '@components/svg/icons';
import { StringUtils } from '@utils';
import { useListsSelector } from '@entities/lists';

interface AddressIndicatorProps {
  address: string;
  hideWatchlist?: boolean;
  hideCollection?: boolean;
  testID?: string;
}

export function AddressIndicator(props: AddressIndicatorProps): JSX.Element {
  const { address, hideCollection, hideWatchlist } = props;
  const { listsOfAddressGroup } = useListsSelector();
  const { watchlist } = useWatchlist();

  const listWithWallet = useMemo(
    () =>
      listsOfAddressGroup.find(
        (list) =>
          list.accounts?.findIndex((account) => account?.address === address) >
          -1
      ),
    [address, listsOfAddressGroup]
  );
  const isWatchlisted = watchlist.findIndex((a) => a.address === address) > -1;

  if (listWithWallet && !hideCollection) {
    return (
      <View style={styles.container}>
        <Text
          fontSize={12}
          fontFamily="Inter_500Medium"
          color={COLORS.gray700}
          numberOfLines={1}
        >
          {StringUtils.formatAddress(listWithWallet.name, 4, 4)}
        </Text>
      </View>
    );
  }
  if (isWatchlisted && !hideWatchlist) {
    return <WatchlistIcon />;
  }
  return <></>;
}
