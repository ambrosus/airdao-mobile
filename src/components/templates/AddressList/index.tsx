import React, { useCallback } from 'react';
import {
  FlatList,
  ListRenderItemInfo,
  ScrollViewProps,
  View
} from 'react-native';
import Animated, { SlideInDown, SlideOutRight } from 'react-native-reanimated';
import { WatchlistTabIcon } from '@components/svg/BottomTabIcons';
import { COLORS } from '@constants/colors';
import { verticalScale } from '@utils/scaling';
import { ExplorerAccount } from '@models/Explorer';
import {
  SwipeableWalletItem,
  SwipeableWalletItemProps
} from './components/SwipeableWalletItem';
import { LocalizedRenderEmpty } from '../LocalizedRenderEmpty';

interface EmptyAddressListProps {
  emptyText: string;
}

interface AddressListProps
  extends EmptyAddressListProps,
    Pick<SwipeableWalletItemProps, 'removeType'> {
  data: ExplorerAccount[];
  isPortfolioFlow?: boolean;
  scrollEnabled?: boolean; // default to true
  contentContainerStyle?: ScrollViewProps['contentContainerStyle'];
  renderItem?: (
    args: ListRenderItemInfo<ExplorerAccount>
  ) => React.ReactElement;
  onRefresh?: () => void;
}

export function AddressList(props: AddressListProps): JSX.Element {
  const {
    data,
    isPortfolioFlow = false,
    scrollEnabled = true,
    removeType = 'watchlist',
    contentContainerStyle = {},
    renderItem,
    onRefresh
  } = props;

  const renderWallet = useCallback(
    (args: ListRenderItemInfo<ExplorerAccount>) => {
      if (typeof renderItem === 'function') return renderItem(args);
      return (
        <Animated.View
          key={args.item.address}
          exiting={SlideOutRight}
          entering={SlideInDown}
        >
          <SwipeableWalletItem
            item={args.item}
            idx={args.index}
            isPortfolioFlow={isPortfolioFlow}
            removeType={removeType}
          />
        </Animated.View>
      );
    },
    [isPortfolioFlow, removeType, renderItem]
  );

  return (
    <FlatList
      contentContainerStyle={[
        {
          flexGrow: 1
        },
        contentContainerStyle
      ]}
      scrollEnabled={scrollEnabled}
      data={data}
      renderItem={renderWallet}
      ListEmptyComponent={
        <View style={{ paddingTop: verticalScale(234) }}>
          <LocalizedRenderEmpty
            text={'empty.addresses'}
            icon={<WatchlistTabIcon color={COLORS.neutral300} scale={1.35} />}
          />
        </View>
      }
      showsVerticalScrollIndicator={false}
      onRefresh={onRefresh}
      refreshing={false} // TODO
    />
  );
}
