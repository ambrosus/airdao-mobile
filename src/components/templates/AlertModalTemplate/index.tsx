import { View } from 'react-native';
import { useTranslation } from 'react-i18next';
import { Row, Text } from '@components/base';
import { PrimaryButton, SecondaryButton } from '@components/modular';
import { InfoIcon } from '@components/svg/icons';
import { COLORS } from '@constants/colors';
import { scale } from '@utils';
import { styles } from './styles';

interface AlertModalProps {
  title?: string;
  subTitle?: string;
  buttonsLabels?: string[];
  onApprove: () => void;
  onReject: () => void;
  buttonLabels?: string[];
}

export const AlertModalTemplate = ({
  title = '',
  subTitle = '',
  buttonsLabels = ['', ''],
  onApprove,
  onReject
}: AlertModalProps) => {
  const { t } = useTranslation();

  return (
    <View style={styles.container}>
      <View style={styles.iconContainer}>
        <InfoIcon scale={0.8} />
      </View>
      <Text
        fontSize={scale(20)}
        fontFamily="Inter_700Bold"
        color={COLORS.neutral800}
        align="center"
      >
        {title || t('bridge.transaction.confirmations')}
      </Text>
      {!!subTitle && (
        <Text
          fontSize={15}
          fontFamily="Inter_500Medium"
          color={COLORS.neutral800}
          align="center"
        >
          {subTitle}
        </Text>
      )}
      <Row width="100%" justifyContent={'space-around'}>
        {buttonsLabels?.[0] && (
          <SecondaryButton style={styles.reject} onPress={onReject}>
            <Text color={COLORS.brand600}>
              {buttonsLabels?.[0] || t('button.cancel')}
            </Text>
          </SecondaryButton>
        )}
        {buttonsLabels?.[1] && (
          <PrimaryButton style={styles.approve} onPress={onApprove}>
            <Text color={COLORS.neutral0}>
              {buttonsLabels?.[1] || t('button.processed')}
            </Text>
          </PrimaryButton>
        )}
      </Row>
    </View>
  );
};
