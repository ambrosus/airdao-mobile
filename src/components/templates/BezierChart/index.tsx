import { useCallback, useRef } from 'react';
import { View } from 'react-native';
import { LineGraph, GraphPoint } from 'react-native-graph';
import { Button, Row, Text } from '@components/base';
import { COLORS } from '@constants/colors';
import { SelectionDot } from './SelectionDot';
import { styles } from './styles';

interface Interval {
  text: string;
  value: any;
}

interface BezierChartProps {
  data: GraphPoint[];
  strokeColor?: string;
  axisLabelColor?: string;
  axisColor?: string;
  intervals?: Interval[];
  selectedInterval?: Interval;
  onPointSelected?: (point?: GraphPoint) => unknown;
  onIntervalSelected?: (interval: Interval) => unknown;
}

export function BezierChart(props: BezierChartProps): JSX.Element {
  const {
    strokeColor = COLORS.black,
    intervals = [],
    selectedInterval,
    data,
    onPointSelected,
    onIntervalSelected
  } = props;
  const gestureStarted = useRef(false);

  const renderInterval = (interval: Interval, idx: number) => {
    const onPress = () => {
      if (typeof onIntervalSelected === 'function') {
        onIntervalSelected(interval);
      }
    };
    return (
      <Button
        testID="Chart_Interval"
        key={interval.value}
        onPress={onPress}
        style={idx > 0 ? styles.interval : {}}
      >
        <Text
          color={
            selectedInterval?.value === interval.value
              ? COLORS.brand600
              : COLORS.neutral300
          }
          fontSize={13}
          fontFamily="Inter_600SemiBold"
        >
          {interval.text}
        </Text>
      </Button>
    );
  };

  const _onPointSelected = useCallback(
    (point?: GraphPoint) => {
      if (typeof onPointSelected === 'function' && gestureStarted.current) {
        onPointSelected(point);
      }
    },
    [onPointSelected]
  );

  const onGestureEnd = useCallback(() => {
    _onPointSelected(undefined);
    gestureStarted.current = false;
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [data, gestureStarted.current]);

  return (
    <View>
      <LineGraph
        testID="Bezier_Chart"
        style={styles.lineGraph}
        points={data}
        animated={true}
        color={strokeColor}
        enablePanGesture={true}
        panGestureDelay={300}
        onGestureStart={() => (gestureStarted.current = true)}
        onGestureEnd={onGestureEnd}
        onPointSelected={_onPointSelected}
        SelectionDot={SelectionDot}
        verticalPadding={21}
      />
      {intervals.length > 0 && (
        <>
          {/* <Spacer value={verticalScale(21)} /> */}
          <Row alignItems="center" justifyContent="center">
            {intervals.map(renderInterval)}
          </Row>
        </>
      )}
    </View>
  );
}
