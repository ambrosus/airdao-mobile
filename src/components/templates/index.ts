export * from './AccountActions';
export * from './AccountList';
export * from './AddressIndicator';
export * from './AddWalletToList';
export * from './AMBPriceHistory';
export * from './AddressList';
export * from './WalletItem';
export * from './BezierChart';
export * from './PortfolioPerformance';
export * from './ReceiveFunds';
export * from './BottomSheetEditCollection';
export * from './BottomSheetRemoveAddressFromCollection';
export * from './BottomSheetCreateRenameGroup';
export * from './BottomSheetSharePortfolio';
export * from './BottomSheetWalletCreateOrImport';
export * from './BottomSheetAddWalletToList';
export * from './NotificationSettings';
export * from './SearchAddress';
export * from './ExplorerAccount';
export * from './BarcodeScanner';
export * from './StakingPoolList';
export * from './TokenPicker';
export * from './TransactionDetails';
export * from './StatusBar';
export * from './LocalizedRenderEmpty';
export * from './WalletPicker';
export * from './WalletTransactionsAndAssets';
export * from './ExplorerAccount/BridgeTransaction';
export * from './BottomSheetChoseNetworks';
