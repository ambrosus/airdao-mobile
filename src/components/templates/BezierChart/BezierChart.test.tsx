import React from 'react';
import { render } from '@testing-library/react-native';
import { BezierChart } from '@components/templates';
import { COLORS } from '@constants/colors';

jest.mock('victory-native', () => {
  return {
    VictoryChart: jest.fn(),
    VictoryTheme: {},
    VictoryLine: jest.fn(),
    VictoryAxis: jest.fn()
  };
});

describe('BezierChart', () => {
  const data = [
    { x: 1, y: 2 },
    { x: 2, y: 3 },
    { x: 3, y: 5 },
    { x: 4, y: 4 },
    { x: 5, y: 5.5 },
    { x: 5, y: 6.5 }
  ];
  it('renders with custom stroke color', () => {
    render(<BezierChart data={data} strokeColor={COLORS.jungleGreen} />);
  });

  it('renders a large dataset', () => {
    const largeData = [
      { x: 1, y: 2 },
      { x: 2, y: 3 },
      { x: 3, y: 4 },
      { x: 4, y: 5 },
      { x: 5, y: 6 },
      { x: 6, y: 7 },
      { x: 7, y: 8 },
      { x: 8, y: 9 },
      { x: 9, y: 10 },
      { x: 10, y: 11 },
      { x: 11, y: 12 },
      { x: 12, y: 13 },
      { x: 13, y: 14 },
      { x: 14, y: 15 },
      { x: 15, y: 16 },
      { x: 16, y: 17 },
      { x: 17, y: 18 },
      { x: 18, y: 19 },
      { x: 19, y: 20 },
      { x: 20, y: 21 },
      { x: 21, y: 22 },
      { x: 22, y: 23 },
      { x: 23, y: 24 },
      { x: 24, y: 25 },
      { x: 25, y: 26 },
      { x: 26, y: 27 },
      { x: 27, y: 28 },
      { x: 28, y: 29 },
      { x: 29, y: 30 },
      { x: 30, y: 31 },
      { x: 31, y: 32 },
      { x: 32, y: 33 },
      { x: 33, y: 34 },
      { x: 34, y: 35 },
      { x: 35, y: 36 },
      { x: 36, y: 37 },
      { x: 37, y: 38 },
      { x: 38, y: 39 },
      { x: 39, y: 40 },
      { x: 40, y: 41 },
      { x: 41, y: 42 },
      { x: 42, y: 43 },
      { x: 43, y: 44 },
      { x: 44, y: 45 },
      { x: 45, y: 46 },
      { x: 46, y: 47 },
      { x: 47, y: 48 },
      { x: 48, y: 49 },
      { x: 49, y: 50 }
    ];

    const tree = render(
      <BezierChart
        data={largeData}
        strokeColor={COLORS.jungleGreen}
        axisLabelColor={COLORS.sapphireBlue}
        axisColor={COLORS.silver}
        height={400}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
