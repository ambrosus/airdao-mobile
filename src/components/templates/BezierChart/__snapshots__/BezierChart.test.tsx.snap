// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BezierChart renders a large dataset 1`] = `null`;
